### ISSUE TITLE
Coordinates must be finite numbers error

### ISSUE THREAD

**Issue by elias75015** [iTowns/itowns#1537]
## Your Environment

* Version used:
* Browser Name and version:
* Operating System and version (desktop or mobile):
* Link to your project:

## Context
 The function this.getGlobeView().controls.pickGeoPosition(coords) returns a new Coordinates('EPSG:4978', pickedPosition).as('EPSG:4326');

Sometimes, the coordinates returned are too long for proj4.
Then, the error (see screenshot below) "Coordinates must be finite numbers" is displayed.

![Screenshot from 2020-12-09 19-15-53](https://user-images.githubusercontent.com/6764800/101671943-7edd5e00-3a55-11eb-8c77-2bdfe86a36cf.png)

## Steps to Reproduce (for bugs)

Reproduced with the geoportal-itowns-extension mousePosition widget

On this sample : https://jsfiddle.net/user/ignfgeoportail/fiddles/3/
1. open the dev console
2. choose "Géographiques" under the "Systèmes de référence"
3. change the units
4. move the mouse on the globe --> the error should be displayed onto the console

In a full iTowns sample, try this code under an "onMouseMove" listener : 

```
    var coords = getGlobeView().eventToViewCoords(mouseEvent);
    return getGlobeView().controls.pickGeoPosition(coords);
```

## Possible Cause/Fix/Solution

As suggested here (https://github.com/proj4js/proj4js/issues/288), maybe try to check how long are the coordinates, or/and parseFloat to a fixed number when creating the Coordinates ?

**Comment by mgermerie** [comment]
@elias75015, I could not reproduce the bug by adding 
```
var coords = view.eventToViewCoord(mouseEvent);
return view.controls.pickGeoPosition(coords);
```
under an "onMouseMove" listener in iTowns [3d map example](http://www.itowns-project.org/itowns/examples/index.html#view_3d_map).

In #1645, I added the fix you suggested (rounding coords that are passed to proj4), but I was unable to test whereas it fixes the issue or not.
Would you happen to know another way of reproducing the error ?

**Comment by Sam-Scolari** [comment]
@mgermerie I just ran into this error when attempting to load the 3d map example on mobile (android) through an in-app webview. I did however notice that loading that same example through chrome worked fine on my phone.

**Comment by elias75015** [comment]
I can't reproduce the bug with itowns 2.30, whatever the browser

**Comment by gchoqueux** [comment]
thanks @elias75015 

@Sam-Scolari you use which itowns version?

**Comment by Sam-Scolari** [comment]
Latest: 2.32.0

The issue seems to come and go for me. Loading the 3d map example on the itowns website throws the error the first time until I reload the page by tapping the 3d map example link on the left side are then it loads fine. But, I hosted my own code on my own site and I didn't encounter the error at all.

**Comment by gchoqueux** [comment]
> The issue seems to come and go for me. Loading the 3d map example on the itowns website throws the error the first time until I reload the page by tapping the 3d map example link on the left side are then it loads fine. But, I hosted my own code on my own site and I didn't encounter the error at all.

You have loading issue or Proj4 message error `Coordinates must be finite numbers error`?

**Comment by Sam-Scolari** [comment]
Proj4 message error Coordinates must be finite numbers error

**Comment by gchoqueux** [comment]
we note this mobile issue, but we can't fix it now.

**Comment by mgermerie** [comment]
@Sam-Scolari would you be able to test if the fix in #1645 works for you ? As I mentioned, I found no way of reproducing the bug on desktop.
